js/bootstrap-select.js

```diff
@@ -2951,6 +2951,10 @@
 
       this.$searchbox.on('input propertychange', function () {
         var searchValue = that.$searchbox[0].value;
+        var isWhitespace = /^\s*$/.test(searchValue);
+        if (!isWhitespace) {
+          searchValue = searchValue.replace(/^\s+|\s+$/g, '');
```

Comment:
I think we're probably okay to just use `searchValue = searchValue.trim();` instead.

Reply:
@caseyjhol 
thank you very much.
Since some users sometimes enter full-width spaces, so I have added support for trimming all half-width spaces and full-width spaces.

Reply:
Ah gotcha. Maybe just mention this in a brief comment above this line.

Reply:
I have added a comment. thank you.